### DISCUSSION TITLE
Changing probe mode into montor

### DISCUSSION THREAD

**Discussion by onurbi** [nzymedefense/nzyme#603] [Questions]
Hi Lenart,

I'm using successfully an Alfa USB stick.
Switching to monitor and back to managed works perfect starting and stopping nzyme.

I am interested, how you realized the switching in Java. Could you please tell me the source file name where I can see the way how you did it? I can Read java code, but probably used the wrong search string crawling through the sources.

Regards,

Onurbi

**Comment by lennartkoopmann** [comment]
It's simply calling what you configured in your `channel_hop_command`. This way you keep maximum flexibility.

**Comment by onurbi** [comment]
Ok, but "iwconfig {interface} channel {channel}" does not switch into monitor mode.
At least 
```
ip link set {interface} down
iw {interface} set monitor none
ip link set {interface} up
```
is neccesary.  So I assume, that monitor mode seems not a precondition for nzyme.

**Reply by lennartkoopmann** [reply]
I apologize, I misread the question.

You are right, the `channel_hop_command` is only used to switch channels. 

By default, nzyme uses the underlying `libpcap` functionality to put the device into monitor mode (`pcap_set_rfmon()`). In some environments this will not work and you can set `skip_enable_monitor` to true and enable monitor mode manually using other tools.

**Comment by onurbi** [comment]
Did you make the experience that using the alternate command "iw [options] dev <devname> set channel <channel> [NOHT|HT20|HT40+|HT40-|5MHz|10MHz|80MHz]" does help to get really all packets using the parameter "80MHz"?

Sometimes I discovered to get a DHCP DISCOVER or ACK, sometimes only DHCP Request for instance. The four EAPOLs I get almost every time.

I see. pcap_set_rfmon makes sense!
I looked for the string "pcap_set_rfmon" in all source files, but for no avail.
Would you mind to tell me the source file name?
